rchttp/client.go

```diff
@@ -159,6 +159,10 @@ func (c *Client) Do(ctx context.Context, req *http.Request) (*http.Response, err
 		return c.backoff(doer, err, ctx, c.HTTPClient, req, errors.New("Bad server status"))
 	}
 
+	if c.ExponentialBackoff && resp.StatusCode == http.StatusConflict {
```

Comment:
can you factor out `if c.ExponentialBackoff {` and put both status code checks inside, please